stackslib/src/chainstate/nakamoto/staging_blocks.rs

```diff
@@ -593,8 +595,15 @@ impl<'a> NakamotoStagingBlocksTx<'a> {
         signing_weight: u32,
         obtain_method: NakamotoBlockObtainMethod,
     ) -> Result<(), ChainstateError> {
-        self.execute("UPDATE nakamoto_staging_blocks SET data = ?1, signing_weight = ?2, obtain_method = ?3 WHERE consensus_hash = ?4 AND block_hash = ?5",
-                    params![&block.serialize_to_vec(), &signing_weight, &obtain_method.to_string(), &block.header.consensus_hash, &block.header.block_hash()])?;
+        self.execute("UPDATE nakamoto_staging_blocks SET data = ?1, signing_weight = ?2, obtain_method = ?3, processed_time = ?4 WHERE consensus_hash = ?5 AND block_hash = ?6",
```

Comment:
The processing time should only be set when the block is _processed_, not stored.  The processing happens in the call to `NakamotoChainState::append_block()`.  The processing time should remain `0` until the block has been processed.

Reply:
I see - so just revert this change? Is my understanding correct that `store_block_if_better` (which calls this `replace_block` fn) is only called before `append_block`?

Reply:
Yes, and yes.